doc/source/data/key-concepts.rst

```diff
@@ -10,19 +10,20 @@ Key Concepts
 Datasets
 --------
 
-A Dataset consists of a list of Ray object references to *blocks*.
-Each block holds a set of items in either `Arrow table format <https://arrow.apache.org/docs/python/data.html#tables>`__
-or a Python list (for non-tabular data).
-For ML use cases, Datasets also natively supports mixing :ref:`Tensor <datasets_tensor_support>` and tabular data.
+A :term:`Dataset` contains a list of Ray object references to :term:`blocks <Block>`.
+Each block holds a set of items in a `Arrow table <https://arrow.apache.org/docs/python/data.html#tables>`_,
+`pandas DataFrame <https://pandas.pydata.org/docs/reference/frame.html>`_, or Python list.
 Having multiple blocks in a dataset allows for parallel transformation and ingest.
 
-Informally, we refer to:
+For ML use cases, Datasets also natively supports mixing :ref:`Tensor <datasets_tensor_support>` and tabular data.
+
+There are three types of datasets:
 
-* A Dataset with Arrow blocks as a *Tabular Dataset*,
-* A Dataset with Python list blocks as a *Simple Dataset*, and
-* A Tabular Dataset with one or more tensor-type columns as a *Tensor Dataset*.
+* :term:`Simple datasets <Simple Dataset>` -- Datasets that represents a collection of Python objects
+* :term:`Tabular datasets <Tabular Dataset>` -- Datasets that represents columnar data
+* :term:`Tensor datasets <Tensor Dataset>` -- Datasets that represents a collection of ndarrays
```

Comment:
should we remove this? we want to hide this type of information from the user right?

Reply:
I think we should keep it for now.

We mention the terms frequently throughout the documentation, so it might be confusing if we don't define them.